runtime-modules/content-directory/src/lib.rs

```diff
@@ -0,0 +1,1645 @@
+                entities_rc_to_increment_vec.push(entities_rc_to_increment);
+            }
+            appended_entity_values.insert(prop_id, new_value.to_owned());
+        } else {
+            // All required prop values should be provided
+            ensure!(!class_prop.required, ERROR_MISSING_REQUIRED_PROP);
+
+            // Add all missing non required schema prop values as PropertyValue::default()
+            appended_entity_values.insert(prop_id, PropertyValue::default());
+        }
+        Ok(())
+    }
+
+    pub fn perform_entity_property_value_update(
+        class: &Class<T>,
+        entity: &Entity<T>,
+        id: PropertyId,
+        access_level: EntityAccessLevel,
+        new_value: PropertyValue<T>,
+        current_prop_value: &mut PropertyValue<T>,
+    ) -> Result<(EntitiesRcVec<T>, EntitiesRcVec<T>), &'static str> {
+        let mut entities_rc_to_increment_vec = vec![];
+        let mut entities_rc_to_decrement_vec = vec![];
+        // Get class-level information about this property
+        if let Some(class_prop) = class.properties.get(id as usize) {
+            // Ensure class property is unlocked for given actor
+            ensure!(
+                !class_prop.is_locked_from(access_level),
+                ERROR_CLASS_PROPERTY_TYPE_IS_LOCKED_FOR_GIVEN_ACTOR
+            );
+
+            // Validate a new property value against the type of this property
+            // and check any additional constraints like the length of a vector
+            // if it's a vector property or the length of a text if it's a text property.
+            class_prop.ensure_property_value_to_update_is_valid(
+                &new_value,
+                entity.get_permissions().get_controller(),
+            )?;
+
+            // Get unique entity ids to update rc
+            if let (Some(entities_rc_to_increment), Some(entities_rc_to_decrement)) = (
+                new_value.get_involved_entities(),
+                current_prop_value.get_involved_entities(),
+            ) {
+                let (entities_rc_to_decrement, entities_rc_to_increment): (
+                    Vec<T::EntityId>,
+                    Vec<T::EntityId>,
+                ) = entities_rc_to_decrement
+                    .into_iter()
+                    .zip(entities_rc_to_increment.into_iter())
+                    .filter(|(entity_rc_to_decrement, entity_rc_to_increment)| {
+                        entity_rc_to_decrement != entity_rc_to_increment
+                    })
+                    .unzip();
+                entities_rc_to_increment_vec.push(entities_rc_to_increment);
+                entities_rc_to_decrement_vec.push(entities_rc_to_decrement);
+            }
+
+            // Update a current prop value in a mutable vector, if a new value is valid.
+            current_prop_value.update(new_value);
+        }
+        Ok((entities_rc_to_increment_vec, entities_rc_to_decrement_vec))
+    }
+
+    fn get_property_value_vec(
```

Comment:
This is ensure.

Reply:
Done https://github.com/Joystream/joystream/pull/470/commits/712f92141c5be42a53c3063479278fbbc6abe09b